services/roomService/src/client/CoveyTownsSocket.test.ts

```diff
@@ -54,119 +60,274 @@ describe('TownServiceApiSocket', () => {
+    const joinData3 = await apiClient.joinTown({
+      coveyTownID: town.coveyTownID,
+      userName: nanoid(),
+    });
+    const { socket: movementSocket } = TestUtils.createSocketClient(
+      server,
+      joinData.coveySessionToken,
+      town.coveyTownID,
+    );
+    const { socket: messageSocket, messageReceived, playerMoved } = TestUtils.createSocketClient(
+      server,
+      joinData2.coveySessionToken,
+      town.coveyTownID,
+    );
+    const {
+      messageReceived: messageReceived2,
+      playerMoved: playerMoved2,
+    } = TestUtils.createSocketClient(server, joinData3.coveySessionToken, town.coveyTownID);
 
-    const newLocation: UserLocation = {x: 100, y: 100, moving: true, rotation: 'back'};
+    const newLocation: UserLocation = { x: 100, y: 100, moving: true, rotation: 'back' };
     movementSocket.emit('playerMovement', newLocation); // moves first client out of range for proximity chat
     await Promise.all([playerMoved, playerMoved2]); // makes sure that other two clients register this movement
 
     // message will be originating from a mew Player's starting location, which is the same place that the second and third clients are located at
-    const message = TestUtils.createMessageForTesting(MessageType.ProximityMessage, new Player(nanoid())); 
+    const message = TestUtils.createMessageForTesting(
+      MessageType.ProximityMessage,
+      new Player(nanoid()),
+    );
     messageSocket.emit('messageSent', message);
     const [firstMessage, secondMessage] = await Promise.all([messageReceived, messageReceived2]); // It should therefore be received by these two clients
     expect(firstMessage).toMatchObject(message);
     expect(secondMessage).toMatchObject(message);
-    // sadly no way of checking that the first client will NOT receive the message. However, if you modify this test to try 
+    // sadly no way of checking that the first client will NOT receive the message. However, if you modify this test to try
     // and await a messageReceived promise from it, the test will timeout.
   });
+  it('Dispatches direct messages to the specified client in the same town, including the client who sent the message', async () => {
+    const town = await createTownForTesting();
+    await apiClient.joinTown({ coveyTownID: town.coveyTownID, userName: nanoid() });
+    const joinData2 = await apiClient.joinTown({
+      coveyTownID: town.coveyTownID,
+      userName: nanoid(),
+    });
+    const { socket, messageReceived } = TestUtils.createSocketClient(
+      server,
+      joinData2.coveySessionToken,
+      town.coveyTownID,
+    );
+
+    const message = TestUtils.createDirectMessageForTesting(
+      joinData2.currentPlayers[0]._id,
+      joinData2.currentPlayers[1]._id,
+    );
+    // emit the direct message action from socket
+    socket.emit('messageSent', message);
+    // message should be received by the second player
+    const [receivedMessage] = await Promise.all([messageReceived]);
```

Comment:
I think you should make sure that it goes to both players: i.e. when you're making joinData2, i.e. create two socket clients based on the two `apiClient.joinTown` data, set up `messageReceived` Promises for both of them, and make sure they both resolve/that the received messages both match `message`